api/users/client/client.go

```diff
@@ -42,3 +48,227 @@ func (c *Client) GetThread(ctx context.Context, name string) (*pb.GetThreadReply
 func (c *Client) ListThreads(ctx context.Context) (*pb.ListThreadsReply, error) {
 	return c.c.ListThreads(ctx, &pb.ListThreadsRequest{})
 }
+
+// SetupMailbox creates inbox and sentbox threads needed user mail.
+func (c *Client) SetupMail(ctx context.Context) (inbox, sentbox thread.ID, err error) {
```

Comment:
Mailbox concept: _A user must opt in by setting up their mailboxes_. In return they will have two threads. One for inbox and one for a sentbox.

Reply:
The main reason though was to be able to watch only a particular mailbox. I figured ACL _could_ be a factor too, though I'm not totally sure how... so just opted for more flex here. I could be convinced to consolidate.

Reply:
Yep, but they both use the same collection schema. I guess we could do sent and received messages

Reply:
Let's give this a shot, it's the two threads are cumbersome, we could just add a different collection schema to the inbox and start putting sent messages there

Reply:
coming in late here. sorry, just some questions to understand. 

sounds like an inbox cross-application then?

Reply:
Yep, but sorta indirectly. It's more that each user (which may be part of multiple apps using the same API key) gets one mail setup. If a user is using the same identity across multiple dev's apps, this current setup means that the first dev is the custodian of the mail threads. Not sure if that'll cause problems or not. If so, we could namespace the mail threads by the account key or something

Reply:
okay cool, sounds good. let's see!